### DISCUSSION TITLE
Emails rejected with "552 5.2.0 Message contains bare CR"

### DISCUSSION THREAD

**Discussion by MarkDynna** [jstedfast/MailKit#1776] [Q&A] [answered]
We recently switched our application over to MailKit to send emails.  Some of our clients are having their emails rejected from one particular email domain with the error "552 5.2.0 Message contains bare CR and is violating 822.bis section 2.3."  Sometimes removing attachments from the emails works and sometimes it doesn't.

We've implemented the library using VB code and are using the BodyBuilder class to build the message and add attachments.

**Comment by jstedfast** [comment]
What version of MailKit? Can you get a protocol log? (zip the log file so that it preserves newline sequences).

**Reply by MarkDynna** [reply]
This would be on version 4.3 of Mailkit.  I'll reach out to our tech support to get them to send some emails with logging enabled and get that sent to me.

**Reply by MarkDynna** [reply]
I'm attaching the log file produced by MailKit.  Also, the rejection notification the client got from the recipient's domain had the message as an attachment so I've included that as well.

Note that the sender and recipient's email addresses have been replaced in these logs for privacy reasons.
[email.txt](https://github.com/user-attachments/files/16289832/email.txt)
[mailkit log.txt](https://github.com/user-attachments/files/16289833/mailkit.log.txt)

**Reply by MarkDynna** [reply]
The log file from MailKit does appear to show two lines in the XML header of the message that have "bare CR's" at the end of the line (as revealed by Notepad++):
![Bare CR in header](https://github.com/user-attachments/assets/72a26b68-4ea6-4df2-9680-2216de37890d)

I'm not sure if there's any way for us to fix this using the BodyBuilder class since we are not crafting the XML directly?

**Reply by jstedfast** [reply, marked as answer]
Okay, this is a bug in your templates or your code. Those bare carriage returns are part of your text/html content.

MailKit will properly convert LF -> CRLF, but it will not convert CR -> CRLF (because no supported system uses bare CR's as a line ending).

UNIX/Linux/MacOS use bare LF's for line endings and Windows uses CRLF.

**Reply by MarkDynna** [reply]
You're right, it's a bug in the HTML/XML editor we use for users to write the conent of their emails.  That one is a library class too, so I'll have to see if they've fixed it.
